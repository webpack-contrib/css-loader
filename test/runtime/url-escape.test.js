const urlEscape = require('../../src/runtime/url-escape');

describe('escape', () => {
  it('should escape url', () => {
    expect(urlEscape(true)).toMatchSnapshot();
    expect(urlEscape('image.png')).toMatchSnapshot();
    expect(urlEscape('"image.png"')).toMatchSnapshot();
    expect(urlEscape("'image.png'")).toMatchSnapshot();
    expect(urlEscape('image other.png')).toMatchSnapshot();
    expect(urlEscape('"image other.png"')).toMatchSnapshot();
    expect(urlEscape("'image other.png'")).toMatchSnapshot();
    expect(urlEscape('image"other.png')).toMatchSnapshot();
    expect(urlEscape('image\nother.png')).toMatchSnapshot();

    expect(urlEscape('image.png', true)).toMatchSnapshot();
    expect(urlEscape("'image other.png'", true)).toMatchSnapshot();
    expect(urlEscape('"image other.png"', true)).toMatchSnapshot();
  });
});
