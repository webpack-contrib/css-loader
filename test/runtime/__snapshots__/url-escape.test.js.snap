// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`escape should escape url 1`] = `true`;

exports[`escape should escape url 2`] = `"image.png"`;

exports[`escape should escape url 3`] = `"image.png"`;

exports[`escape should escape url 4`] = `"image.png"`;

exports[`escape should escape url 5`] = `"\\"image other.png\\""`;

exports[`escape should escape url 6`] = `"\\"image other.png\\""`;

exports[`escape should escape url 7`] = `"\\"image other.png\\""`;

exports[`escape should escape url 8`] = `"\\"image\\\\\\"other.png\\""`;

exports[`escape should escape url 9`] = `"\\"image\\\\nother.png\\""`;
